pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -235,21 +236,54 @@ def setSuns(self, Ee, cells=None):
         """
         if cells is None:
             if np.isscalar(Ee):
-                for pvc in self.pvcells:
+                new_pvcells = range(self.numberCells)  # new list of cells
+                old_pvcells = dict.fromkeys(self.pvcells)  # same as set(pvcells)
+                for cell_id, pvcell in enumerate(self.pvcells):
+                    if old_pvcells[pvcell] is None:
+                        new_pvcells[cell_id] = copy(pvcell)
+                        old_pvcells[pvcell] = new_pvcells[cell_id]
+                    else:
+                        new_pvcells[cell_id] = old_pvcells[pvcell]
+                self.pvcells = new_pvcells
+                for pvc in set(self.pvcells):
                     pvc.Ee = Ee
             elif np.size(Ee) == self.numberCells:
-                for pvc, Ee_idx in zip(self.pvcells, Ee):
-                    pvc.Ee = Ee_idx
+                self.pvcells = copy(self.pvcells)  # copy list first
+                for cell_idx, Ee_idx in enumerate(Ee):
+                    # gh34: make new objects as needed from copies
```

Comment:
and can we add a `TODO` comment here to optimize this later. it `pvmod.setSuns(Ee=0.1)` maybe seldom used, but I think it will unnecessarily add a lot of new cells, that could possibly be identical. Probably the exact same technique as used else where to catalog the unique cells and their new copies could be applied here in the future. You could put that in the `TODO` too.